### DISCUSSION TITLE
remove window variable #13367

### DISCUSSION THREAD

**Discussion by aaronhuggins** [denoland/deno#13373] [General]
Why would the `window` variable be a candidate for removal? I seem to be missing some kind of feedback or discussion. @ry opened #13367 and I'm wondering what the driving thought is for the issue, and I didn't want to clutter the issue with pointless discussion.

Thanks!

**Comment by lucacasonato** [comment]
I have updated the description on the issue to give some background @aaronhuggins :-)

**Reply by aaronhuggins** [reply]
Wow, I hate the reason for it. But I understand the reasoning and it makes sense.

As devs using the Deno runtime, I suppose we can always polyfill `window` on `globalThis` or at the top-level for those times when the behavior is desirable on a per-project basis.

The only thing I enjoyed about the existence of `window` is that I could write code that would run as expected in both browser and Deno. I'm going to miss that, but it was a small creature comfort and not a huge need.

**Reply by lucacasonato** [reply]
@aaronhuggins Yeah, it is very unfortunate. As a best practice, you should probably use `globalThis` or `self` instead of `window` anyway, so your code also runs in workers. We actually have a lint rule for that: https://lint.deno.land/#no-window-prefix

**Reply by aaronhuggins** [reply]
@lucacasonato  Let's all just please kill Internet Explorer. 😄 🔪 🥇
